### PR TITLE
Add draft release notes for 2.10.17

### PR DESCRIPTION
**Pull request by bunchesofdonald**
Release notes for 2.10.17

### CODE REVIEW

File: RELEASE-NOTES.md [hunk 1]
```diff
@@ -1,5 +1,27 @@
 # Prefect Release Notes
 
+## Release 2.10.17
```
Comment (serinamarie) [inline RELEASE-NOTES.md:3]: Making we can take this opportunity to celebrate our new contributors since we have more than usual?

Otherwise we could highlight our new and improved tutorial!

File: RELEASE-NOTES.md [hunk 2]
```diff
@@ -1,5 +1,27 @@
...
+
+### Enhancements
+- Update PrefectClient to follow redirects by default — https://github.com/PrefectHQ/prefect/pull/9988
+- Add the CustomWebhookNotificationBlock for general purpose notification — https://github.com/PrefectHQ/prefect/pull/9547
```
Comment (serinamarie) [inline RELEASE-NOTES.md:7]:
```suggestion
- Add the `CustomWebhookNotificationBlock` for general purpose notification — https://github.com/PrefectHQ/prefect/pull/9547
```

File: RELEASE-NOTES.md [hunk 3]
```diff
@@ -1,5 +1,27 @@
...
+
+### Fixes
+- Resolve bug that could throw an OSError if too many files were being scanned for flows — https://github.com/PrefectHQ/prefect/pull/10014
+
+### Documentation
+- Fix broken link in deployments UX page — https://github.com/PrefectHQ/prefect/pull/10013
+- Document --work-queue / -q arguments to `worker start` command — https://github.com/PrefectHQ/prefect/pull/10027
```
Comment (serinamarie) [inline RELEASE-NOTES.md:14]:
```suggestion
- Document `--work-queue / -q` arguments to `worker start` command — https://github.com/PrefectHQ/prefect/pull/10027
```

File: RELEASE-NOTES.md [hunk 4]
```diff
@@ -1,5 +1,27 @@
...
+- Add a link to join Club 42 to the Community page — https://github.com/PrefectHQ/prefect/pull/9927
```
Comment (serinamarie) [inline RELEASE-NOTES.md:15]:
```suggestion
- Add link to join Club 42 to Community page — https://github.com/PrefectHQ/prefect/pull/9927
```

File: RELEASE-NOTES.md [hunk 5]
```diff
@@ -1,5 +1,27 @@
 # Prefect Release Notes
 
+## Release 2.10.17
+
+### Enhancements
+- Update PrefectClient to follow redirects by default — https://github.com/PrefectHQ/prefect/pull/9988
+- Add the CustomWebhookNotificationBlock for general purpose notification — https://github.com/PrefectHQ/prefect/pull/9547
+
+### Fixes
+- Resolve bug that could throw an OSError if too many files were being scanned for flows — https://github.com/PrefectHQ/prefect/pull/10014
```
Comment (desertaxle) [inline RELEASE-NOTES.md:10]:
```suggestion
- Limit the number of files concurrently opened by `prefect deploy` when searching for flows — https://github.com/PrefectHQ/prefect/pull/10014
```

File: RELEASE-NOTES.md [hunk 6]
```diff
@@ -1,5 +1,27 @@
...
+
+### Documentation
+- Fix broken link in deployments UX page — https://github.com/PrefectHQ/prefect/pull/10013
+- Document --work-queue / -q arguments to `worker start` command — https://github.com/PrefectHQ/prefect/pull/10027
+- Add a link to join Club 42 to the Community page — https://github.com/PrefectHQ/prefect/pull/9927
+- Update the tutorial to be more succinct and purposeful  — https://github.com/PrefectHQ/prefect/pull/9940
```
Comment (serinamarie) [inline RELEASE-NOTES.md:16]:
```suggestion
- Update Prefect tutorial to be more succinct and purposeful  — https://github.com/PrefectHQ/prefect/pull/9940
```

File: RELEASE-NOTES.md [hunk 7]
```diff
@@ -1,5 +1,27 @@
...
+
+### Enhancements
+- Update PrefectClient to follow redirects by default — https://github.com/PrefectHQ/prefect/pull/9988
+- Add the CustomWebhookNotificationBlock for general purpose notification — https://github.com/PrefectHQ/prefect/pull/9547
+
+### Fixes
+- Resolve bug that could throw an OSError if too many files were being scanned for flows — https://github.com/PrefectHQ/prefect/pull/10014
+
+### Documentation
+- Fix broken link in deployments UX page — https://github.com/PrefectHQ/prefect/pull/10013
```
Comment (desertaxle) [inline RELEASE-NOTES.md:13]:
```suggestion
- Fix broken link to `prefect-docker` documentation on the deployments UX page — https://github.com/PrefectHQ/prefect/pull/10013
```

File: RELEASE-NOTES.md (hunk 6 above)
Comment (desertaxle) [inline RELEASE-NOTES.md]: This PR seems like a good one to use for the highlight section.

File: RELEASE-NOTES.md (hunk 5 above)
Comment (serinamarie) [inline RELEASE-NOTES.md]:
```suggestion
- Resolve bug that could throw `OSError` if too many files were being scanned for flows — https://github.com/PrefectHQ/prefect/pull/10014
```

File: RELEASE-NOTES.md (hunk 2 above)
Comment (desertaxle) [inline RELEASE-NOTES.md]:
```suggestion
- New `CustomWebhookNotificationBlock` for triggering custom webhooks in response to flow run state changes — https://github.com/PrefectHQ/prefect/pull/9547
```

File: RELEASE-NOTES.md [hunk 8]
```diff
@@ -1,5 +1,27 @@
 # Prefect Release Notes
 
+## Release 2.10.17
+
+### Enhancements
+- Update PrefectClient to follow redirects by default — https://github.com/PrefectHQ/prefect/pull/9988
```
Comment (serinamarie) [inline RELEASE-NOTES.md:6]:
```suggestion
- Update Prefect client to follow redirects by default — https://github.com/PrefectHQ/prefect/pull/9988
```

Comment (serinamarie) [reply RELEASE-NOTES.md:6]: or `PrefectClient` but not used as often in our notes

File: RELEASE-NOTES.md [hunk 9]
```diff
@@ -1,5 +1,27 @@
...
+- Add `CustomWebhookNotificationBlock` for general purpose notification — https://github.com/PrefectHQ/prefect/pull/9547
+
+### Fixes
+- Limit the number of files concurrently opened by `prefect deploy` when searching for flows — https://github.com/PrefectHQ/prefect/pull/10014
```
Comment (pleek91) [inline RELEASE-NOTES.md:10]:
```suggestion
- Limit the number of files concurrently opened by `prefect deploy` when searching for flows — https://github.com/PrefectHQ/prefect/pull/10014
- Fix js error when running prefect in an unsecure context (http vs https) — https://github.com/PrefectHQ/prefect-ui-library/pull/1501
```

Comment (serinamarie) [reply RELEASE-NOTES.md:10]: @pleek91 Can you rephrase 
"Fix js error when running prefect in an unsecure context (http vs https)" as what that means for the user?

File: RELEASE-NOTES.md (hunk 8 above)
Comment (pleek91) [inline RELEASE-NOTES.md:6]:
```suggestion
- Update PrefectClient to follow redirects by default — https://github.com/PrefectHQ/prefect/pull/9988
- Always show checkboxes rather than animating them on hover — https://github.com/PrefectHQ/prefect-ui-library/pull/1490
```

File: RELEASE-NOTES.md (hunk 1 above)
Comment (bunchesofdonald) [reply RELEASE-NOTES.md:3]: I added a highlight for the tutorial.

File: RELEASE-NOTES.md (hunk 9 above)
Comment (serinamarie) [reply RELEASE-NOTES.md:15]: What page does it cause to break in the Prefect UI?

Comment (pleek91) [reply RELEASE-NOTES.md:15]: Just the ui in general, its not specific to a single page

File: RELEASE-NOTES.md [hunk 10]
```diff
@@ -1,5 +1,33 @@
 # Prefect Release Notes
 
+## Release 2.10.17
+
+### Improved Prefect Tutorial
+
+Unleash the full potential of Prefect with the [newly improved tutorial](https://docs.prefect.io/2.10.17/tutorial/). Elevate your learning experience and embark on a transformative journey that equips you with the skills to orchestrate and observe all of your workflows.
```
Comment (billpalombi) [inline RELEASE-NOTES.md:7]:
```suggestion
Prefect's documentation has a [new and improved tutorial](https://docs.prefect.io/2.10.17/tutorial/). The tutorial focuses on the key elements and features of Prefect. You can find content related to more advanced features or less common use cases in the [Guides](https://docs.prefect.io/2.10.17/guides/) section
```

File: RELEASE-NOTES.md [hunk 11]
```diff
@@ -1,5 +1,33 @@
...
+
+### Enhancements
+- Update Prefect client to follow redirects by default — https://github.com/PrefectHQ/prefect/pull/9988
+- Always show checkboxes rather than animating them on hover — https://github.com/PrefectHQ/prefect-ui-library/pull/1490
```
Comment (billpalombi) [inline RELEASE-NOTES.md:11]:
```suggestion
- Always show checkboxes on list items, rather than animating them on hover — https://github.com/PrefectHQ/prefect-ui-library/pull/1490
```

File: RELEASE-NOTES.md (hunk 10 above)
Comment (serinamarie) [inline RELEASE-NOTES.md]:
```suggestion
We're excited to announce an enhanced tutorial, designed to showcase the latest features in Prefect. With the introduction of concepts like work pools and workers, alongside an interactive deployment CLI, this tutorial reflects the elevated experience that these new features bring.

Follow along as we simplify the entire process, providing a seamless experience from creating a flow to having a worker execute your flow. By the end of this tutorial, you'll have gained a comprehensive understanding of Prefect's core concepts and be equipped to harness their power effectively. You'll learn to:

1. Createa Flow: Learn how to define and structure your workflows using Python code, with Prefect's intuitive syntax.
2. Add Tasks: Discover how to incorporate tasks into your flow, enabling the execution of specific actions or computations.
3. Create a Work Pool: Explore the concept of work pools, which allow you to distribute tasks across multiple workers efficiently.
4. Deploy a Worker: Gain hands-on experience in deploying a worker, a vital component that executes tasks within your flow.
5. Deploy the Flow: Learn how to deploy your flow, ensuring it is readily available for execution on your worker.
6. Run the Flow on the Worker: Run your flow seamlessly on the worker.
```

**Review by billpalombi — APPROVED** [review]
lgtm

**Comment by fatjoni** [comment]
The randomUUID bug still persists at the 2.10.17 version.
